modules/mod_acl_user_groups/support/acl_user_groups_checks.erl

```diff
@@ -212,6 +212,110 @@ acl_is_allowed(#acl_is_allowed{action=Action, object=ModuleName}, Context) when
+                true -> 30;
+                false -> 0
+            end
+    end.
+
+
+% Things - check only their content group
+privacy_check(20, false, Id, Context) ->
+    % Same user group, but not a person -- apply collaboration group semantics
+    privacy_check(30, false, Id, Context);
+privacy_check(30, false, Id, Context) ->
+    is_collab_group_member(m_rsc:p_no_acl(Id, content_group_id, Context), Context);
+privacy_check(40, false, Id, Context) ->
+    is_collab_group_manager(m_rsc:p_no_acl(Id, content_group_id, Context), Context);
+% People - check their memberships *and* their content group
+privacy_check(20, true, _Id, #context{acl=#aclug{user_groups=[]}}) ->
+    false;
+privacy_check(20, true, Id, #context{acl=#aclug{user_groups=UGs}} = Context) ->
+    case has_user_groups(Id, Context) of
+        [] -> false;
+        Other ->
+            Other1 = Other -- [ m_rsc:rid(acl_user_group_members, Context) ],
+            lists:any(fun(UgId) -> lists:member(UgId, UGs) end, Other1)
+    end
+    orelse privacy_check(30, true, Id, Context);
+privacy_check(30, true, _Id, #context{acl=#aclug{collab_groups=[]}}) ->
+    false;
+privacy_check(30, true, Id, #context{acl=#aclug{collab_groups=CGs}} = Context) ->
+    case has_collab_groups(Id, Context) of
+        [] -> false;
+        Other ->
+            lists:any(fun(UgId) -> lists:member(UgId, CGs) end, Other)
+            orelse lists:member(m_rsc:p_no_acl(Id, content_group_id, Context), CGs)
+    end
+    orelse lists:member(m_rsc:p_no_acl(Id, content_group_id, Context), CGs);
+privacy_check(40, true, _Id, #context{acl=#aclug{collab_groups=[]}}) ->
+    false;
+privacy_check(40, true, Id, #context{user_id=UserId} = Context) ->
+    case m_edge:subjects(UserId, hascollabmanager, Context) of
+        [] -> false;
+        ManagerOf ->
+            case has_collab_groups(Id, Context) of
+                [] -> false;
+                Other ->
+                    lists:any(fun(UgId) -> lists:member(UgId, ManagerOf) end, Other)
+                    orelse lists:member(m_rsc:p_no_acl(Id, content_group_id, Context), ManagerOf)
+            end
+    end;
+privacy_check(_Privacy, _IsPerson, _Id, _Context) ->
+    false.
+
+
+is_private_property(email) -> true;
+is_private_property(phone) -> true;
+is_private_property(phone_mobile) -> true;
+is_private_property(phone_alt) -> true;
+is_private_property(address_street_1) -> true;
+is_private_property(address_street_2) -> true;
+is_private_property(address_postcode) -> true;
+is_private_property(address_city) -> true;
+is_private_property(date_start) -> true;
+is_private_property(date_end) -> true;
+is_private_property(location_lat) -> true;
+is_private_property(location_lng) -> true;
+is_private_property(pivot_location_lat) -> true;
+is_private_property(pivot_location_lng) -> true;
+is_private_property(pivot_geocode) -> true;
+is_private_property(pivot_geocode_qhash) -> true;
+is_private_property(_) -> false.
```

Comment:
BTW if a person's name should be private, then I assume it is better to place the whole resource into a different content group.

This privacy setting should be restricted to some details of an otherwise viewable resource.